### PR TITLE
feat(command): `zks config`

### PR DESCRIPTION
**Pull request by ilitteri**
This PR introduces the `zks config` command.

# Description

This command is in charge of managing the configuration the CLI will use for all the other commands. Running `zks config` shows you the available commands:

```
CLI config commands.

Usage: zks config <COMMAND>

Commands:
  edit     Edit an existing config.
  create   Create a new config.
  set      Set the config to use.
  display  Display a config.
  list     List all configs.
  delete   Delete a config.
  help     Print this message or the help of the given subcommand(s)

Options:
  -h, --help  Print help
```

You can create multiple configs as "profiles" to be able to use the tool with for example various servers, or wallets without having to use different commands with different flags or have to manually edit one config file each time.

# Commands

## `zks config create`

A command used to create new configurations interactively.

### Usage

```
Create a new config.

Usage: zks config create --name <CONFIG_NAME>

Options:
      --name <CONFIG_NAME>
  -h, --help                Print help
```

## `zks config edit`

Used for editing existing configurations. You can do this with flags to edit specific values or you can edit it interactively.

### Usage

```
Edit an existing config.

Usage: zks config edit [OPTIONS]

Options:
      --name <CONFIG_NAME>
      --l1-rpc-url <L1_RPC_URL>
      --l2-rpc-url <L2_RPC_URL>
      --l2-explorer-url <L2_EXPLORER_URL>
      --l1-explorer-url <L1_EXPLORER_URL>
      --private-key <PRIVATE_KEY>
      --address <ADDRESS>
  -e, --interactively
  -h, --help                               Print help
```

## `zks config set`

Once you've created one or multiple configs it is time to set which one you're going to use. For this, you can use the `zks config set` command which lets you choose interactively, which config among the created ones you want to set; or you can set it by name.

### Usage

```
Set the config to use.

Usage: zks config set [OPTIONS] --name <CONFIG_NAME>

Options:
      --name <CONFIG_NAME>
  -s, --interactively
  -h, --help                Print help
```

## `zks config display`

If you want to see what configuration is set in one specific config profile, use `zks config display` to display a config file by name or choose it interactively among the existing config files.

### Usage

```
Display a config.

Usage: zks config display [OPTIONS] --name <CONFIG_NAME>

Options:
      --name <CONFIG_NAME>
  -s, --interactively
  -h, --help                Print help
```

## `zks config list`

Lists all the existing configurations.

### Usage

```
List all configs.

Usage: zks config list

Options:
  -h, --help  Print help
```

## `zks config delete`

Deletes an existing config. You can either delete by name or choose interactively among the existing configs.

### Usage

```
Delete a config.

Usage: zks config delete [OPTIONS]

Options:
      --name <CONFIG_NAME>
  -d, --interactively
  -h, --help                Print help
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,17 +23,13 @@ Whether you're a developer focused on deploying and interacting with contracts o
 ## Installation
 
 ```
-git clone git@github.com:lambdaclass/zksync_era_cli.git
-cd zksync_era_cli
+git clone git@github.com:lambdaclass/zksync_cli.git
+cd zksync_cli
```
Comment (ilitteri) [inline README.md]:
```suggestion
git clone git@github.com:lambdaclass/zksync_era_cli.git
cd zksync_era_cli
```

File: src/commands/config/common/mod.rs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        .join("zks-cli")
+        .join("configs");
+    if !configs_dir_path.exists() {
+        std::fs::create_dir_all(&configs_dir_path)?;
+    }
+    Ok(configs_dir_path)
+}
+
+pub fn config_path(config_name: &str) -> eyre::Result<std::path::PathBuf> {
+    Ok(configs_dir_path()?.join(&format!("{config_name}.toml")))
```
Comment (fborello-lambda) [inline src/commands/config/common/mod.rs]: Running clippy:
```warning: the borrowed expression implements the required traits
  --> src/commands/config/common/mod.rs:50:33
   |
50 | ...th()?.join(&format!("{config_name}.toml")))
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: change this to: `format!("{config_name}.toml")
```
```suggestion
    Ok(configs_dir_path()?.join(format!("{config_name}.toml")))
```

File: src/commands/config/common/mod.rs [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+};
+use dialoguer::{theme::ColorfulTheme, Confirm, Input, Select};
+use eyre::ContextCompat;
+use std::{path::PathBuf, str::FromStr};
+use zksync_ethers_rs::types::H160;
+
+pub const SELECTED_CONFIG_FILE_NAME: &str = ".selected";
+
+pub const DEFAULT_L1_RPC_URL: &str = "http://localhost:8545";
+pub const DEFAULT_L2_RPC_URL: &str = "http://localhost:3030";
```
Comment (fborello-lambda) [inline src/commands/config/common/mod.rs]:
```suggestion
pub const DEFAULT_L2_RPC_URL: &str = "http://localhost:3050";
```

**Review by fborello-lambda — COMMENTED** [review]
Looks good, just one comment, if I'm using a config and i edit the same config, the `set command has to be used in order to apply the changes. They are not applied automatically.

File: src/commands/config/common/mod.rs (hunk 2 above)
Comment (ilitteri) [reply src/commands/config/common/mod.rs]: It's weird, the CI passes and when doing `make cli` this doesn't appear. Nevertheless, it's a good catch.

**Comment by ilitteri** [comment]
> Looks good, just one comment, if I'm using a config and i edit the same config, the `set` command has to be used in order to apply the changes. They are not applied automatically.

This is correct, I'll handle this case asap.

**Comment by ilitteri** [comment]
> Looks good, just one comment, if I'm using a config and i edit the same config, the `set` command has to be used in order to apply the changes. They are not applied automatically.

Done in [ffbc376](https://github.com/lambdaclass/zksync_era_cli/pull/42/commits/ffbc3769af83dc90c4ce2e8f408779143d87181a).

**Review by fborello-lambda — APPROVED** [review]
LGTM

**Pull request merged**